src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -74,8 +90,10 @@ public override FormulaValue Visit(TableNode node, SymbolContext context)
             var args = new FormulaValue[len];
             for (var i = 0; i < len; i++)
```

Comment:
1) We should configure our code analysis to make sure that CheckCancel() appears in every loop, OR
2) We should have a loop-helper class or function that guarantees that CheckCancel() is called during the loop. Then configure code analysis to not allow loops anywhere else in the file.

@lesaltzm Could we use our code analysis tool for either of these?

Reply:
I'd vote for the first.  

But we need to write a custom analyzer for it - so that's a separate task. (Even the 2nd item will still require a custom analyzer).